File: Sources/CountriesFlagsObjC/source/Extensions/UIBezierPath+Plus.m
//
//  UIBezierPath+Plus.m
//  CountriesFlags
//

#import "Extensions/UIBezierPath+Plus.h"

@implementation UIBezierPath (Plus)

+ (UIBezierPath *)plusInSize:(CGSize)size sideWidth:(CGFloat)sideWidth {
    CGFloat xOffset = size.width / 2.0f - sideWidth * 3.0f / 2.0f;
    CGFloat yOffset = size.height / 2.0f - sideWidth * 3.0f / 2.0f;

    UIBezierPath* path = [UIBezierPath bezierPath];
    // top
    [path moveToPoint:CGPointMake(xOffset + sideWidth, yOffset)];
    [path addLineToPoint:CGPointMake(xOffset + sideWidth * 2.0f, yOffset)];
    [path addLineToPoint:CGPointMake(xOffset + sideWidth * 2.0f, yOffset + sideWidth)];
    // right
    [path addLineToPoint:CGPointMake(xOffset + sideWidth * 3.0f, yOffset + sideWidth)];
    [path addLineToPoint:CGPointMake(xOffset + sideWidth * 3.0f, yOffset + sideWidth * 2.0f)];
    [path addLineToPoint:CGPointMake(xOffset + sideWidth * 2.0f, yOffset + sideWidth * 2.0f)];
    // bottom
    [path addLineToPoint:CGPointMake(xOffset + sideWidth * 2.0, yOffset + sideWidth * 3.0f)];
    [path addLineToPoint:CGPointMake(xOffset + sideWidth, yOffset + sideWidth * 3.0f)];
    [path addLineToPoint:CGPointMake(xOffset + sideWidth, yOffset + sideWidth * 2.0f)];
    // left
    [path addLineToPoint:CGPointMake(xOffset, yOffset + sideWidth * 2.0f)];
    [path addLineToPoint:CGPointMake(xOffset, yOffset + sideWidth)];
    [path addLineToPoint:CGPointMake(xOffset + sideWidth, yOffset + sideWidth)];
    [path closePath];

    return path;
}

@end

File: Sources/CountriesFlags/Extensions/UIBezierPath+Plus.swift
//
//  UIBezierPath+Plus.swift
//  CountriesFlags
//

import UIKit

extension UIBezierPath {

    static func plus(in size: CGSize, with sideWidth: CGFloat) -> UIBezierPath {
        let xOffset = size.width / 2.0 - sideWidth * 3.0 / 2.0
        let yOffset = size.height / 2.0 - sideWidth * 3.0 / 2.0

        let path = UIBezierPath()
        // top
        path.move(to: CGPoint(x: xOffset + sideWidth, y: yOffset))
        path.addLine(to: CGPoint(x: xOffset + sideWidth * 2.0, y: yOffset))
        path.addLine(to: CGPoint(x: xOffset + sideWidth * 2.0, y: yOffset + sideWidth))
        // right
        path.addLine(to: CGPoint(x: xOffset + sideWidth * 3.0, y: yOffset + sideWidth))
        path.addLine(to: CGPoint(x: xOffset + sideWidth * 3.0, y: yOffset + sideWidth * 2.0))
        path.addLine(to: CGPoint(x: xOffset + sideWidth * 2.0, y: yOffset + sideWidth * 2.0))
        // bottom
        path.addLine(to: CGPoint(x: xOffset + sideWidth * 2.0, y: yOffset + sideWidth * 3.0))
        path.addLine(to: CGPoint(x: xOffset + sideWidth, y: yOffset + sideWidth * 3.0))
        path.addLine(to: CGPoint(x: xOffset + sideWidth, y: yOffset + sideWidth * 2.0))
        // left
        path.addLine(to: CGPoint(x: xOffset, y: yOffset + sideWidth * 2.0))
        path.addLine(to: CGPoint(x: xOffset, y: yOffset + sideWidth))
        path.addLine(to: CGPoint(x: xOffset + sideWidth, y: yOffset + sideWidth))
        path.close()

        return path
    }
}
